File: module-services/src/main/java/eu/geekhome/services/configurable/StringField.java
package eu.geekhome.services.configurable;

import eu.geekhome.services.localization.Resource;

public class StringField extends FieldDefinition<String> {

    public StringField(String name, Resource hint, int maxSize, Validator<String>... validators) {
        super(name, hint, maxSize, String.class, new StringFieldBuilder(), validators);
    }
}

File: modules/domain/src/main/java/eu/geekhome/domain/configurable/StringField.kt
package eu.geekhome.domain.configurable

import eu.geekhome.data.fields.FieldType
import eu.geekhome.data.localization.Resource

class StringField(
    name: String,
    hint: Resource,
    maxSize: Int,
    initialValue: String,
    vararg validators: Validator<String?>
) : FieldDefinition<String>(
    FieldType.String, name, hint, maxSize, initialValue, String::class.java,
    StringFieldBuilder(), null, *validators)